app/configurator/components/field.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import { fireEvent, render } from "@testing-library/react";
+import { getD3TimeFormatLocale } from "../../locales/locales";
+import { TimeInput } from "./field";
+
+describe("TimeInput", () => {
+  const expectedValue = "2020-05-24";
+  const setup = ({ isOptional, id }: { isOptional: boolean; id: string }) => {
+    let currentValue = "";
+
+    const root = render(
+      <TimeInput
+        id={id}
+        label={id}
+        value={currentValue}
+        timeFormat="%Y-%m-%d"
+        formatLocale={getD3TimeFormatLocale("en")}
+        isOptional={isOptional}
+        onChange={(e) => (currentValue = e.currentTarget.value)}
+      />
+    );
+
+    return { node: root.getByLabelText(id), getValue: () => currentValue };
```

Comment:
Nit: I think it is a little bit convoluted to use the functionality of the component to set a local variable. I think a spy on onChange would be a bit clearer, in the tests you would directly see that the onChange behavior is tested.